### ISSUE TITLE
Suggestions/glide typing don't input in some apps

### ISSUE THREAD

**Issue by tsiflimagas** [florisboard/florisboard#1009]
#### Short description
Writing in WordReference app, I had noticed that using glide typing or suggestions couldn't input any words in the app's input field. I mention suggestions because it's been a long time since I detected this bug, but forgot to report it😐 Also I'm writing "apps" in the title supposing there are other apps with which FlorisBoard exhibits the same behaviour too (actually I think I had found one more, but can't remember it)

I just noticed that swipe to delete doesn't work either.
#### Steps to reproduce
1. Go to WordReference app
2. Use glide trying (or suggestions too if being on an older version)
3. Notice words not getting input

#### Environment information
- FlorisBoard Version: 0.3.13-beta04 (currently)
- Install Source: GitHub
- Device: Xiaomi redmi note 8
- Android: 11, custom rom

**Comment by patrickgold** [comment]
Same as #1247, the search box of the WordReference app is incorrectly configured (inputType is NULL, which indicates raw editor):

```
2022-05-12 19:37:15.134 24089-24089/dev.patrickgold.florisboard.debug I/FlorisImeService: onStartInputView() - restarting=false info=android.view.inputmethod.EditorInfo
    packageName: com.wordreference
    ---
    inputType: class=0x00000000 variation=0x00000000 flags=(none)
    imeOptions: action=IME_ACTION_SEARCH flags=IME_FLAG_NAVIGATE_NEXT|IME_FLAG_NO_EXTRACT_UI
    privateImeOptions: (null)
    ---
    actionId: 0x0000007b
    actionLabel: (null)
    contentMimeTypes: []
    extras: [android.support.text.emoji.emojiCompat_metadataVersion=8,android.support.text.emoji.emojiCompat_replaceAll=false]
    hintLocales: (null)
    hintText: Search the dictionary
    ---
    initialCapsMode: flags=(none)
    initialSelStart: 0
    initialSelEnd: 0
```

As I also mentioned in linked issue though is that I will work on still providing a bare bone glide typing experience even for raw editors, so this bug should be fixed by that then. If you want you can send the devs of the WordReference app above log and explanation, so they can fix this issue.

**Comment by patrickgold** [comment]
Above linked commit/PR fixes the issue. What I did is I changed the way I detect rich editors a bit, so the WordReference search box (and other apps that have this exact same bug) is now detected as "rich" and glide typing works. Additionally for "real" raw editors (e.g. Termux) glide typing is now completely disabled, this means no glide trail that does nothing anymore. I cannot reliably provide a good glide typing experience for such terminal apps, because I am just missing too much info about surrounding symbols and text.

Even if FlorisBoard works now with WordReference I will still contact the devs of WordReference, so they can fix this bug and other Android keyboards can work properly in their app again.

Will be released in 0.3.16-beta03. Thanks for reporting!
